agent/exec/container/executor.go

```diff
@@ -76,6 +76,7 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 			NanoCPUs:    int64(info.NCPU) * 1e9,
 			MemoryBytes: info.MemTotal,
 		},
+		Addr: info.Swarm.NodeAddr,
```

Comment:
I'm wondering if this should be autodetected by the manager instead of being pushed up by the agent.

The agent doesn't always know its externally-reachable address. And if its address changes, the agent won't know, but the manager will.